src/TestUtils/src/UITest.Appium.AI/HelperExtensions.cs

```diff
@@ -0,0 +1,825 @@
+		{
+			var chatClient = app.GetChatClient();
+
+			var screenshot = app.Screenshot() ?? throw new InvalidOperationException("Failed to get screenshot");
+			var screenshotBytes = BinaryData.FromBytes(screenshot);
+
+			var messages = new List<ChatMessage>
+			{
+				new SystemChatMessage(VerifyPrompt),
+				new UserChatMessage(
+					ChatMessageContentPart.CreateTextPart(prompt),
+					ChatMessageContentPart.CreateImagePart(screenshotBytes, "image/png"))
+			};
+
+			var options = new ChatCompletionOptions
+			{
+				Temperature = Temperature,
+				MaxOutputTokenCount = MaxOutputTokenCount,
+				FrequencyPenalty = 0,
+				PresencePenalty = 0,
+			};
+
+			ChatCompletion completion = await chatClient.CompleteChatAsync(messages, options);
+
+			if (completion.Content != null && completion.Content.Count > 0)
+			{
+				string result = completion.Content[0].Text;
+				VerifyData? verifyData = JsonSerializer.Deserialize<VerifyData>(SanitizeResponse(result));
+
+				if (verifyData is not null)
+					return verifyData.Answer;
+			}
+
+			return false;
+		}
+
+		/// <summary>
+		/// Verifies that the provided screenshot matches the reference screenshot using AI.
+		/// </summary>
+		/// <param name="app">Represents the main gateway to interact with an app.</param>
+		/// <param name="app">The application instance where the screenshots are taken.</param>
+		/// <param name="snapshot">The current screenshot to be verified.</param>
+		/// <param name="referenceSnapshot">The reference screenshot to compare against.</param>
+		/// <param name="prompt">An optional AI prompt to assist in the verification process.</param>
+		/// <returns>Returns true if the snapshots are equals.</returns>
+		public static async Task<bool> VerifyScreenshotWithAI(this IApp app, BinaryData snapshot, BinaryData referenceSnapshot, string? prompt = null)
+		{
+			var chatClient = app.GetChatClient();
+
+			var messages = new List<ChatMessage>
+			{
+				new SystemChatMessage(VerifyScreenshotPrompt),
+				new UserChatMessage(
+					ChatMessageContentPart.CreateTextPart(prompt ?? "Compare the images"),
+					ChatMessageContentPart.CreateImagePart(snapshot, "image/png"),
+					ChatMessageContentPart.CreateImagePart(referenceSnapshot, "image/png"))
+			};
+
+			var options = new ChatCompletionOptions
+			{
+				Temperature = Temperature,
+				MaxOutputTokenCount = MaxOutputTokenCount,
+				FrequencyPenalty = 0,
+				PresencePenalty = 0,
+			};
+
+			ChatCompletion completion = await chatClient.CompleteChatAsync(messages, options);
+
+			if (completion.Content != null && completion.Content.Count > 0)
+			{
+				string result = completion.Content[0].Text;
+				VerifyScreenshotData? verifyScreenshotData = JsonSerializer.Deserialize<VerifyScreenshotData>(SanitizeResponse(result));
+
+				if (verifyScreenshotData is not null)
+					return verifyScreenshotData.AreEquals;
+			}
+
+			return false;
+		}
+
+		public static async Task ExecuteWithAI(this IApp app, string prompt)
```

Comment:
At the moment, only support basic actions like tap, enter or clear text etc. If it becomes more than a POC, we must implement more complex actions such as scrolling, swipes, etc.